### PR TITLE
Fix failing Windows CI tests

### LINKED ISSUE #231: Failing Windows CI tests on main branch
We have two consistently failing tests in Windows CI on the `main` branch:

Example: https://build.osrfoundation.org/job/ign_fuel-tools-pr-win/112/

[FuelClientTest.CachedModel](https://build.osrfoundation.org/job/ign_fuel-tools-pr-win/112/testReport/junit/(root)/FuelClientTest/CachedModel/)
[LocalCacheTest.AllModels](https://build.osrfoundation.org/job/ign_fuel-tools-pr-win/112/testReport/junit/(root)/LocalCacheTest/AllModels/)

-----

### FuelClientTest.CachedModel

https://github.com/ignitionrobotics/ign-fuel-tools/blob/c47d85f00419d4a684f1955e56fb4404a060f77d/src/FuelClient_TEST.cc#L617-L625

```
[ RUN      ] FuelClientTest.CachedModel
5: [36m[Dbg] [C:\Jenkins\workspace\ign_fuel-tools-pr-win\ws\ign-fuel-tools\src\FuelClient_TEST.cc:48] [m[36mCreating local model in [[m[36mC:\Jenkins\workspace\ign_fuel-tools-pr-win\ws\build\ignition-fuel_tools8[m[36m][m[36m
5: [m[33m[Wrn] [C:\Jenkins\workspace\ign_fuel-tools-pr-win\ws\ign-fuel-tools\src\LocalCache.cc:105] [m[33mServer directory does not exist [[m[33mC:\Jenkins\workspace\ign_fuel-tools-pr-win\ws\build\ignition-fuel_tools8\test_cache\fuel.ignitionrobotics.org[m[33m]
5: [mC:\Jenkins\workspace\ign_fuel-tools-pr-win\ws\ign-fuel-tools\src\FuelClient_TEST.cc(621): error: Value of: result
5:   Actual: false
5: Expected: true
5: C:\Jenkins\workspace\ign_fuel-tools-pr-win\ws\ign-fuel-tools\src\FuelClient_TEST.cc(622): error: Expected equality of these values:
5:   ResultType::FETCH_ALREADY_EXISTS
5:     Which is: 4-byte object <05-00 00-00>
5:   result.Type()
5:     Which is: 4-byte object <07-00 00-00>
5: C:\Jenkins\workspace\ign_fuel-tools-pr-win\ws\ign-fuel-tools\src\FuelClient_TEST.cc(624): error: Expected equality of these values:
5:   common::joinPaths(common::cwd(), "test_cache", "localhost:8007", "alice", "models", "My Model", "3")
5:     Which is: "C:\\Jenkins\\workspace\\ign_fuel-tools-pr-win\\ws\\build\\ignition-fuel_tools8\\test_cache\\localhost8007\\alice\\models\\My Model\\3"
5:   path
5:     Which is: ""
```

### LocalCacheTest.AllModels

https://github.com/ignitionrobotics/ign-fuel-tools/blob/c47d85f00419d4a684f1955e56fb4404a060f77d/src/LocalCache_TEST.cc#L248-L258

```15: [ RUN      ] LocalCacheTest.AllModels
15: [36m[Dbg] [C:\Jenkins\workspace\ign_fuel-tools-pr-win\ws\ign-fuel-tools\src\LocalCache_TEST.cc:47] [m[36mCreating 6 local models in [[m[36mC:\Jenkins\workspace\ign_fuel-tools-pr-win\ws\build\ignition-fuel_tools8[m[36m][m[36m
15: [m[36m[Dbg] [C:\Jenkins\workspace\ign_fuel-tools-pr-win\ws\ign-fuel-tools\src\LocalCache_TEST.cc:99] [m[36mCreating 3 local models in [[m[36mC:\Jenkins\workspace\ign_fuel-tools-pr-win\ws\build\ignition-fuel_tools8[m[36m][m[36m
15: [m[33m[Wrn] [C:\Jenkins\workspace\ign_fuel-tools-pr-win\ws\ign-fuel-tools\src\LocalCache.cc:105] [m[33mServer directory does not exist [[m[33mC:\Jenkins\workspace\ign_fuel-tools-pr-win\ws\build\ignition-fuel_tools8\test_cache\fuel.ignitionrobotics.org[m[33m]
15: [mC:\Jenkins\workspace\ign_fuel-tools-pr-win\ws\ign-fuel-tools\src\LocalCache_TEST.cc(258): error: Expected: (uniqueNames.end()) != (uniqueNames.find( "http://localhost:8001/alice/models/am1")), actual: 8-byte object <50-F6 94-ED 10-02 00-00> vs 8-byte object <50-F6 94-ED 10-02 00-00>
15: [  FAILED  ] LocalCacheTest.AllModels (45 ms)```

### PR DESCRIPTION
**Pull request by mjcarroll**
There were two tests that have been failing as reported in #231.

Both of these are due to URI/Path changes that happened a little while back in ignition-common

### CODE REVIEW

File: src/LocalCache_TEST.cc [hunk 1]
```diff
@@ -242,13 +242,16 @@ TEST_F(LocalCacheTest, AllModels)
...
   {
+    std::cerr << iter->Identification().UniqueName() << std::endl;
     uniqueNames.insert(iter->Identification().UniqueName());
     ++iter;
   }
   EXPECT_EQ(9u, uniqueNames.size());
 
-  EXPECT_NE(uniqueNames.end(), uniqueNames.find(
-      "http://localhost:8001/alice/models/am1"));
+  auto toFind = ignition::common::joinPaths(
```
Comment (ahcorde) [inline src/LocalCache_TEST.cc:251]: uhmm, I don't think this is a good idea, on windows you will use `backslash` to join the URI strings

**Review by ahcorde — CHANGES_REQUESTED** [review]
These changes also break some tests on Linux 

```bash
         5 - UNIT_FuelClient_TEST (Failed)
  	 15 - UNIT_LocalCache_TEST (Failed)
  	 17 - UNIT_ModelIdentifier_TEST (Failed)
```

File: src/LocalCache_TEST.cc (hunk 1 above)
Comment (mjcarroll) [reply src/LocalCache_TEST.cc:251]: I actually came up from looking at the `AllWorlds` test below, which had mostly the same logic, but was passing: https://github.com/ignitionrobotics/ign-fuel-tools/blob/3c901105bb90bfdca918ea0d89715ab6fd11bb88/src/LocalCache_TEST.cc#L385-L391

I don't know that it's the 100% correct solution, but I want to get CI green so I can do some heavier cleaning.

**Comment by mjcarroll** [comment]
> These changes also break some tests on Linux

Still a work in progress, I'll switch back to draft.

**Pull request closed without merging**